apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/DiagnosisUtilV2.java

```diff
@@ -178,86 +177,146 @@ private static void addCodingToCodeableConcept(
    * @param substitution The methods to retrive diagnosis information all follow a similar pattern.
    *     This value is used to substitute into that pattern when looking up the specific method to
    *     retrive information with.
-   * @param claim Passed as an Object because there is no top level `Claim` class that claims derive
-   *     from
+   * @param codes The mapping of diagnosis codes by their property name and respective value
+   * @param codeVersions The mapping of diagnosis code versions by their property name and
+   *     respective value
+   * @param presentOnAdms The mapping of diagnosis "PresentOnAdmissionCode" codes by their property
+   *     name and respective value
    * @param ccw CCW Codebook value that represents which "PresentOnAdmissionCode" is being used.
    *     Example: {@link CcwCodebookVariable#CLM_POA_IND_SW5}
    * @param label One or more labels to use when mapping the diagnosis.
    * @return a {@link Diagnosis} or {@link Optional#empty()}
    */
   public static Optional<Diagnosis> extractDiagnosis(
-      String substitution, Object claim, Optional<CcwCodebookInterface> ccw, DiagnosisLabel label) {
-
+      String substitution,
+      Map<String, Optional<String>> codes,
+      Map<String, Optional<Character>> codeVersions,
+      Optional<Map<String, Optional<Character>>> presentOnAdms,
```

Comment:
We might want to consider just passing in a `Map<String, Optional<Character>>` rather than an `Optional<Map<String, Optional<Character>>>` fir `presentOnAdms`.  The callers can just pass `Map.of()` in the cases where they don't have a real map. The  case that does have a map (InpatientClaimTransformerV2) could just pass it in directly.

That would make the code using `presentOnAdms` a lot simpler.